### DISCUSSION TITLE
recording time between events

### DISCUSSION THREAD

**Discussion by spikey1973** [micropython/micropython#10373] [ESP32] [answered]
hello to all, 

I am fairly new to programming and have been at building a program to get my wemos esp32 s2 mini board to do as I intend it to. 
most of it is fairly basic (never the less I still strugg with that is half the fun, not) but now i am a far outside my comfort zone as i don't even know where to start or where to look. 

I wish to record time be tween two events.. 
event 1) = start timer (hardware) as part of a for loop. 
event 2a) = end timer upon switch value to go from 0 to 1  -> print the recorded time to a file.
event 2b) = or for the timer to run out a preset time period (7000ms) -> print a string "ran out" to that file instead. 

the essence of the problem is the time recording.. printing that information to file is of a later issue. 

now i looked at the class Timers, but i can't seem to see hoe i can use this class to denote a time period. 
I also googled examples but none of the examples i found relate.. generally it is used delay an event to happen for a certain amount of set time but that is different. 

I hope someone here has some experience and can point me towards the right path / some realating examples. 

Kind greats 

Matthieu

**Comment by davefes** [comment, marked as answer]
https://docs.micropython.org/en/latest/library/time.html

time.ticks

**Reply by davefes** [reply]
I have seen examples on the old forum:

So, search for "timer ticks"

**Reply by spikey1973** [reply]
Hello davefas, 

Thank you so much for putting me on this track.. I will go and search for the ticks on the old forum # nolime ;) 

Happy new year!!!

**Reply by scruss** [reply]
more specifically, you likely want to use the [time.ticks_diff()](https://docs.micropython.org/en/v1.19.1/library/time.html#time.ticks_diff) function to calculate the difference between two ticks values. It handles timer rollover properly.
